tests/run-make/reset-codegen-1/rmake.rs

```diff
@@ -31,8 +31,8 @@ fn main() {
         ("multi-output", Some("asm,obj")),
     ];
     for (output_file, emit) in flags {
-        fs::remove_file(output_file).unwrap_or_default();
+        std::fs::remove_file(rmake_out_path(output_file)).unwrap_or_default();
```

Comment:
Problem: this logic is wrong for the multi-output case. When you request multiple output artifacts via `--emit=asm,obj`, rustc has to produce at least two separate output artifact files. This mandates that the different kinds of output artifacts cannot be in the same file, and so they are actually named `{output_file}.s` and `{output_file}.o` respectively. But in the one-output-kind case, rustc can omit the extension and use the output filename via `-o` as provided. I missed this during the review of the original port, sorry!

This is why you observe

> As demonstrated, not commenting out the last case results in `the file in path ""/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/reset-codegen-1/reset-codegen-1/rmake_out/multi-output"" could not be removed.: Os { code: 2, kind: NotFound, message: "No such file or directory" }`

Reply:
Suggestion: we could just pull the multi-output case out of the loop and be dealt with separately.

Reply:
A very astute observation! The good thing about this test porting project is how we sometimes need to go back to already written tests to reflect improvements in the API, which is good for catching stuff like this.

Changes pushed.

Reply:
Sweet, looks like it does work. @rustbot review